### PR TITLE
Improved Datatables View (and some other minor improvements as well)

### PR DESCRIPTION
**Pull request by malcolmflint**
During my time at Yahoo/Oath as an Intern, I made improvements to the datatables implementation in the table visualization. Most notably I was able to improve the scrolling as well as add an export to csv button that will export the filtered table. I also improved the speed of rendering the visualization in part by removing the last metric column if there are no metrics given.

### CODE REVIEW

File: metadata.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (xrmx) [inline metadata.json]: What is this file needed for?

File: superset/views/core.py [hunk 2]
```diff
@@ -61,7 +61,6 @@
 USER_MISSING_ERR = __("The user seems to have been deleted")
```
Comment (xrmx) [inline superset/views/core.py]: I'd drop all the changes to this file unless there's a good reason you need them

File: superset/assets/package.json [hunk 3]
```diff
@@ -54,6 +54,11 @@
     "datatables-bootstrap3-plugin": "^0.5.0",
     "datatables.net": "^1.10.13",
     "datatables.net-bs": "^1.10.12",
+    "datatables.net-buttons": "^1.3.1",
+    "datatables.net-buttons-bs": "^1.3.1",
+    "font-awesome": "^4.6.3",
```
Comment (xrmx) [inline superset/assets/package.json]: I think we dropped this in favour of the fab shipped one.

File: superset/assets/visualizations/filter_box.jsx [hunk 4]
```diff
@@ -10,19 +10,25 @@ import { TIME_CHOICES } from './constants';
...
-  instantFiltering: PropTypes.bool,
   filtersChoices: PropTypes.object,
   onChange: PropTypes.func,
   showDateFilter: PropTypes.bool,
   datasource: PropTypes.object.isRequired,
+  showLabel: PropTypes.bool,
+  showMetricNumber: PropTypes.bool,
+  origSelectedValues: PropTypes.object,
+  instantFiltering: PropTypes.bool,
+  sliceId: PropTypes.number,
```
Comment (xrmx) [inline superset/assets/visualizations/filter_box.jsx]: Are we actually using sliceId for something?

File: metadata.json (hunk 1 above)
Comment (malcolmflint) [reply metadata.json]: My bad, that one snuck it's way in. I'll remove it. Just a sec

File: superset/views/core.py (hunk 2 above)
Comment (malcolmflint) [reply superset/views/core.py]: the reason to go to appbuilder.sm.auth_view.login (line 2231) is so that if someone were to use OAuth and define their own security manager it doesn't get stuck in a redirect loop. With default authentication, I believe it works exactly the same.

File: superset/assets/package.json (hunk 3 above)
Comment (malcolmflint) [reply superset/assets/package.json]: dt buttons?

File: superset/assets/visualizations/filter_box.jsx (hunk 4 above)
Comment (malcolmflint) [reply superset/assets/visualizations/filter_box.jsx]: It appears not. That was in there when I began work on the filter_box, so I added to prop_types for consistency...Let me know and I'll remove it from the entire file.

Comment (xrmx) [reply superset/assets/visualizations/filter_box.jsx]: If sliceId is not used no need to keep it.

File: superset/assets/package.json (hunk 3 above)
Comment (xrmx) [reply superset/assets/package.json]: The comment is for font-awesome

File: superset/views/core.py (hunk 2 above)
Comment (xrmx) [reply superset/views/core.py]: Then that's fine

File: superset/assets/package.json (hunk 3 above)
Comment (malcolmflint) [reply superset/assets/package.json]: ah, I would be happy to remove the line provided it won't break anything. I'll do that now

File: superset/assets/visualizations/filter_box.jsx (hunk 4 above)
Comment (malcolmflint) [reply superset/assets/visualizations/filter_box.jsx]: I will remove that now
